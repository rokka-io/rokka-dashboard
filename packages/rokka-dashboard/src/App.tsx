import React, { FunctionComponent, memo } from 'react';
import { HashRouter as Router, Route } from 'react-router-dom';
import { routes } from './pages/routes';
import { GlobalStyle } from 'rokka-dashboard-ui-base';
import { withAuthRequired, WithUserProps } from './hoc/withAuthRequired';
import { withState } from './hoc/withState';
import { BaseLayout } from './layouts/BaseLayout/BaseLayout';
import { AppUser } from './state';

interface AppProps extends WithUserProps {
  showSidebar: boolean;
}

const App: FunctionComponent<AppProps> = memo(({ user, showSidebar }) => {
  return (
    <>
      <GlobalStyle />
      <Router>
        <BaseLayout user={user} showSidebar={showSidebar}>
          <>
            {routes.map(({ name, path, exact, PageComponent }) => {
              return <Route key={name} path={path} exact={exact} component={PageComponent} />;
            })}
          </>
        </BaseLayout>
      </Router>
    </>
  );
});

export const AppContainer = withState<{}, { user: AppUser | null; showSidebar: boolean }>(
  withAuthRequired(App),
  ({ user, showSidebar }) => ({
    user,
    showSidebar
  })
);
