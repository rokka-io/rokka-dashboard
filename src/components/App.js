import React, { PureComponent } from 'react'
import { Route } from 'react-router'
import { HashRouter } from 'react-router-dom'
import state, { subscribe } from '../state'
import Dashboard from './Dashboard'
import Images from './Images'
import Bills from './Bills'
import Signup from './Signup'
import Stacks from './Stacks'
import NewStack from './NewStack'
import '../scss/main.scss'

class App extends PureComponent {
  constructor(props) {
    super(props)

    this.state = state

    this.unsubscribe = subscribe(state => {
      this.setState(state)
    })
  }

  componentWillUnmount() {
    this.unsubscribe()
  }

  render() {
    return (
      <HashRouter>
        <div className="App">
          <Route path="/" exact render={props => <Dashboard router={props} {...this.state} />} />
          <Route path="/images" render={props => <Images router={props} {...this.state} />} />
          <Route path="/bills" render={props => <Bills router={props} {...this.state} />} />
          <Route path="/signup" render={props => <Signup router={props} {...this.state} />} />
          <Route path="/stacks" render={props => <Stacks router={props} {...this.state} />} />
          <Route path="/new-stack" render={props => <NewStack router={props} {...this.state} />} />
        </div>
      </HashRouter>
    )
  }
}

export default App
