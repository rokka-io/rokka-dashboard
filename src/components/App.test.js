import React from 'react'
import App from '../../src/components/App'
import renderer from 'react-test-renderer'

test('App does render', () => {
  const component = renderer.create(
    <App>
      <span>Test</span>
    </App>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
