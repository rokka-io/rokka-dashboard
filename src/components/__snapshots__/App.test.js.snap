// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App does render 1`] = `
<div
  className="App"
>
  <div
    className="rka-login-page"
  >
    <div
      className="rka-login-container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-5"
        >
          <div
            className="rka-login-brand"
          >
            <i
              className="rka-header-logo mb-lg"
            />
            <h2
              className="txt-white rka-h2"
            >
              Web images done right.
            </h2>
            <p
              className="rka-login-brand-powered"
            >
              Powered by Liip.
            </p>
          </div>
        </div>
        <div
          className="col-md-7"
        >
          <div
            className="rka-login-form-container"
          >
            <form
              onSubmit={[Function]}
            >
              <div
                className="rka-form-group"
              >
                <label
                  className="rka-label"
                  htmlFor="organization"
                >
                  Organization
                </label>
                <input
                  className="rka-input-txt"
                  defaultValue={null}
                  id="organization"
                  name="organization"
                  onChange={[Function]}
                  type="text"
                />
              </div>
              <div
                className="rka-form-group"
              >
                <label
                  className="rka-label"
                  htmlFor="apiKey"
                >
                  Api Key
                </label>
                <input
                  className="rka-input-txt"
                  defaultValue={null}
                  id="apiKey"
                  name="apiKey"
                  onChange={[Function]}
                  type="password"
                />
              </div>
              <button
                className="rka-button rka-button-brand mt-sm"
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
